src/lib/sort/sort-header.ts

```diff
@@ -46,6 +46,7 @@ import {getMdSortHeaderNotContainedWithinMdSortError} from './sort-errors';
   styleUrls: ['sort-header.css'],
   host: {
     '(click)': '_sort.sort(this)',
+    '[attr.aria-sort]': '_getAriaSortAttribute()',
```

Comment:
What does the reader announce when we have both the `aria-sort` and the description?

Could you try this in NVDA and VoiceOver as well?

Reply:
Here's JAWS with different options (with/without aria-sort/visually-hidden div)

With aria-sort, without visually-hidden: `change sorting for userid button sorted ascending`
Without aria-sort, with visually-hidden: `change sorting for userid button sorted by userid ascending`
With both: `change sorting for userid button sorted ascending sorted by userid ascending sorted ascending`

Checking out NVDA and VoiceOver...

Reply:
NVDA was simliar, wasn't able to get VoiceOver to play nice.